### PR TITLE
🌱 CRS should return errors when trying to apply custom resources

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This is a short term fix until we have #3400 or some other similar solution.

/assign @sedefsavas

### CODE REVIEW

**Comment by vincepri** [comment]
/retitle 🌱 CRS should return errors when trying to apply custom resources

**Comment by sedefsavas** [comment]
Thanks for the PR @vincepri. 
We can rethink if we want to reconcile in errors like Patch or getClient in #3400, but this temporary solution LGTM.

File: exp/addons/controllers/clusterresourceset_controller.go [hunk 1]
```diff
@@ -211,6 +209,11 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 				conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.WrongSecretTypeReason, clusterv1.ConditionSeverityWarning, err.Error())
 			} else {
 				conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.RetrievingResourceFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
+
+				// Continue without adding the error to the aggregate if we can't find the resource.
```
Comment (andrewsykim) [inline exp/addons/controllers/clusterresourceset_controller.go:213]: I'm missing a lot of context here so sorry for the newbie question:

I would think the Secret/ConfigMap resource not existing would still result in an error and a retry, what's the reason for skipping it here?

Comment (vincepri) [reply exp/addons/controllers/clusterresourceset_controller.go:213]: That would be tackled in a different PR, which will add watches on both ConfigMaps/Secrets we're interested in. When a new secret gets created, we get notified and we reconcile the linked CRS resources

Comment (andrewsykim) [reply exp/addons/controllers/clusterresourceset_controller.go:213]: I see, even in that case, if during reconcile the resource got deleted and was no longer available shouldn't we err and retry?

Comment (sedefsavas) [reply exp/addons/controllers/clusterresourceset_controller.go:213]: If resource gets deleted, no need to retry until it is recreated again and we will be watching create and update events on ConfigMaps/Secrets. Do you agree?

Comment (andrewsykim) [reply exp/addons/controllers/clusterresourceset_controller.go:213]: I think so! Should we add this check when the ConfigMap/Secret watch is added? Or does this watch already exist? Maybe I missed it.

Comment (sedefsavas) [reply exp/addons/controllers/clusterresourceset_controller.go:213]: That change is tracked here: https://github.com/kubernetes-sigs/cluster-api/issues/3397
Will open up a PR soon.

Comment (andrewsykim) [reply exp/addons/controllers/clusterresourceset_controller.go:213]: gotcha, thanks!

**Pull request merged**